src/util/FactorUtil.js

```diff
@@ -347,12 +347,18 @@ function (Okta, TimeUtil) {
   };
 
   fn.removeRequirementsFromError = function (responseJSON, policy) {
-    var passwordRequirementsAsString = this.getPasswordComplexityDescription(policy);
+    var passwordRequirementsAsString = fn.getPasswordComplexityDescription(policy);
     if (responseJSON.errorCauses
-        && responseJSON.errorCauses.length > 0
-        && _.isString(responseJSON.errorCauses[0].errorSummary)) {
-      responseJSON.errorCauses[0].errorSummary = responseJSON.errorCauses[0].errorSummary
+        && responseJSON.errorCauses.length > 0) {
+      if (_.isString(responseJSON.errorCauses[0].errorSummary)) {
+        responseJSON.errorCauses[0].errorSummary = responseJSON.errorCauses[0].errorSummary
         .replace(passwordRequirementsAsString, '').trim();
+      }
+      // OIE type response
+      if (Array.isArray(responseJSON.errorCauses[0].errorSummary) && responseJSON.errorCauses[0].errorSummary.length) {
```

Comment:
I am seeing it to be an array --
```
                {
                  "name":"passcode",
                  "label":"Enter password",
                  "secret":true,
                  "messages":{
                    "type":"array",
                    "value":[
                      {
                        "message":"Password requirements were not met. [Password requirements: at least 8 characters, a lowercase letter, an uppercase letter, a number, a symbol, no parts of your username, does not include your first name, does not include your last name. Your password cannot be any of your last 4 passwords. At least 2 hour(s) must have elapsed since you last changed your password.]",
                        "i18n":{
                          "key":"password.passwordRequirementsNotMet",
                          "params":[
                            [
                              "Password requirements: at least 8 characters, a lowercase letter, an uppercase letter, a number, a symbol, no parts of your username, does not include your first name, does not include your last name. Your password cannot be any of your last 4 passwords. At least 2 hour(s) must have elapsed since you last changed your password."
                            ]
                          ]
                        },
                        "class":"ERROR"
                      }
                    ]
                  }
                }
```

Reply:
could you double check with backend engineer to see if it's intentional or mistake from API (with v.s. without `[`)?

Reply:
Yeah, it seems weird that we are getting it this way. Can you double check? Since that changes the logic in this PR as well.

Reply:
Waiting on someone from @cred-okta to answer - 
https://okta.slack.com/archives/C9AJXB5GC/p1598892006006900
https://okta.slack.com/archives/C9AJXB5GC/p1599155536094000

Reply:
https://okta.slack.com/archives/C9AJXB5GC/p1599155536094000 - Plan is to remove what is inside the brackets including the brackets in which case this filtering will not alter anything (added a test case to prove fault tolerance)

Reply:
Left a comment on a backend ticket to _not_ remove entire message instead only the brackets - https://oktainc.atlassian.net/browse/OKTA-317654